compio-runtime/src/runtime/mod.rs

```diff
@@ -76,8 +80,13 @@ impl RuntimeInner {
         let main_id = std::thread::current().id();
         let poll = self.poll;
         let schedule = move |runnable| {
-            runnables.push(runnable);
-            if poll || main_id != std::thread::current().id() {
+            let in_current_thread = main_id == std::thread::current().id();
```

Comment:
I doubt that if `std::thread::current().id()` would slow down enough to counter the benefits of current-thread-optimization at all. It needs more benchmarks.

Reply:
Well I didn't see it in the flamegraph...

Reply:
I've checked the source code and found that the current thread id is cached in a TLS. I think it is not that slow.

Reply:
It's only generated once (which takes time due to syscalls) but after that it's only a single read to TLS storage so it's fine.